pkg/pillar/cmd/downloader/downloader.go

```diff
@@ -330,65 +343,67 @@ func handleCreate(ctx *downloaderContext, objType string,
 		log.Fatalf("handleCreate: No ObjType for %s\n",
 			config.ImageID)
 	}
-	// Start by marking with PendingAdd
-	status := types.DownloaderStatus{
-		ImageID:          config.ImageID,
-		DatastoreID:      config.DatastoreID,
-		Name:             config.Name,
-		ObjType:          objType,
-		IsContainer:      config.IsContainer,
-		RefCount:         config.RefCount,
-		LastUse:          time.Now(),
-		AllowNonFreePort: config.AllowNonFreePort,
-		PendingAdd:       true,
-	}
-	publishDownloaderStatus(ctx, &status)
-
-	// Check if we have space
-	// Update reserved space. Keep reserved until doDelete
-	// XXX RefCount -> 0 should keep it reserved.
-	kb := types.RoundupToKB(config.Size)
-	if !tryReserveSpace(ctx, &status, kb) {
-		errString := fmt.Sprintf("Would exceed remaining space. "+
-			"SizeOfAppImage: %d, RemainingSpace: %d\n",
-			kb, ctx.globalStatus.RemainingSpace)
-		log.Errorln(errString)
+	if status == nil {
+		// Start by marking with PendingAdd
+		status0 := types.DownloaderStatus{
+			ImageID:          config.ImageID,
+			DatastoreID:      config.DatastoreID,
+			Name:             config.Name,
+			ObjType:          objType,
+			IsContainer:      config.IsContainer,
+			RefCount:         config.RefCount,
+			LastUse:          time.Now(),
+			AllowNonFreePort: config.AllowNonFreePort,
+			PendingAdd:       true,
+		}
+		status = &status0
+	} else {
+		status.ImageID = config.ImageID
```

Comment:
Please add comment that this is about resurrecting a refcount=0 object to trigger a download.

Reply:
Sure.